### PR TITLE
Simpler model map

### PR DESCRIPTION
**Pull request by bastbijl**
This is quite big, but what it does is simplify the internal reference catalogue keeps of the model with its includes. It was listed as WEAV-349.

It also solves a bunch of error tickets:
- WEAV-244
- WEAV-285
- WEAV-287
- WEAV-309
- WEAV-310

I certify that:
- [x] The automated build passes on the branch this pull request is from
- [x] My change is documented in the Changelist.md
- [ ] Any functionality additions/changes are documented in the README.md
- [ ] Any functionality additions/changes are _also_ documented [here](https://github.com/weaverplatform/weaver-docs/blob/master/pages/developers/reference/weaver-sdk-js.md "Weaver-docs")
- [x] Any new dockerimages this build depends on are properly tagged on github
- [x] I shall not merge until the travis build is green and the PR is accepted by
  another developer
- [x] I shall promptly fix/reply to comments, as I know PRs are not :fire: and
  forget
- [x] I shall keep my branch updated with any intermediate changes to the
  target branch
- [x] This is a quality PR
- [x] I got to the end of this list

### CODE REVIEW

File: test/WeaverModelQuery.test.coffee [hunk 1]
```diff
@@ -118,12 +119,17 @@ describe 'WeaverModelQuery test', ->
           persons[0].should.be.instanceOf(model.Person)
         )
 
-      it.skip 'handles selectOut correctly', ->
```
Comment (mingushead) [inline test/WeaverModelQuery.test.coffee]: <3

**Comment by mingushead** [comment]
Does this also provide a fix for https://issues.sysunite.com/projects/WEAV/issues/WEAV-342 ?

**Comment by bastbijl** [comment]
No, this was most likely already solved, either by some connector update or some recent patch in this sdk-js.

File: test/WeaverModel.test.coffee [hunk 2]
```diff
@@ -205,6 +213,26 @@ describe 'WeaverModel test', ->
...
+        ).then((node)->
+          expect(node).to.be.instanceOf(model.Person)
+          node.relation('worksIn').first().load()
+        ).then((node)->
+          expect(node).to.be.instanceOf(model.Office)
+          node.relation('placedIn').first().load()
+        ).then((node)->
+          expect(node).to.be.instanceOf(model.Country)
+        )
+
```
Comment (mingushead) [inline test/WeaverModel.test.coffee]: <3

File: test/WeaverModelRelation.test.coffee [hunk 3]
```diff
@@ -65,3 +65,23 @@ describe 'Weaver Model relation test', ->
...
+    p.relation('worksIn').add(o)
+    o.relation('placedIn').add(c)
+    p.save().then(->
+      model.Person.load('p')
+    ).then((node)->
+      loadedNode = node
+      expect(loadedNode).to.be.instanceOf(model.Person)
+      assert.throws((->loadedNode.relation('worksIn').load(model.Country)))
+
+      loadedNode.relation('worksIn').load(model.Office)
```
Comment (mingushead) [inline test/WeaverModelRelation.test.coffee]: is the `model.Office` argument required here now, or does it default to a model-specified constructor?

Comment (bastbijl) [reply test/WeaverModelRelation.test.coffee]: I'm not sure, I'll add this to the test if it works without

Comment (bastbijl) [reply test/WeaverModelRelation.test.coffee]: It is necessary to give this argument.

Comment (mingushead) [reply test/WeaverModelRelation.test.coffee]: Ok, I guess this is outside the scope of this pr anyway

**Comment by mingushead** [comment]
can't find the _approve_ button, but looks good to me!

**Comment by malamili** [comment]
Looks good to me @bastbijl! How much hurry does this PR has? Im suspecting @gvdent to have a comment here or there before we merge it

**Comment by gvdent** [comment]
Could you check (possibly with the help of @mingushead) whether the developer documentation at: https://github.com/weaverplatform/weaver-docs/blob/master/pages/developers/reference/weaver-sdk-js.md#step-3-nest-models-to-describe-complex-structures is still correct after this change?

**Comment by bastbijl** [comment]
It wasnt before the change. And is still not.

**Comment by mingushead** [comment]
Following on from @gvdent , that documentation is _hella_ out-of-date. Maybe even _crazy_ out-of-date. 
On the upside, it was still _dayum_ out-of-date before @bastbijl 's current pr. This documentation was for a poc Model implementation that I did in March 2017 (The Weaver.Model that we have all come to know and love appeared in a published version around December 2017, I think).

**Pull request merged**